rskj-core/src/main/java/co/rsk/mine/MinerServer.java

```diff
@@ -26,15 +26,28 @@
 public interface MinerServer {
 
     void start();
+    void stop();
+
+    boolean isRunning();
 
     SubmitBlockResult submitBitcoinBlock(String blockHashForMergedMining, co.rsk.bitcoinj.core.BtcBlock bitcoinMergedMiningBlock);
 
+    boolean generatePrivateBlock();
+
+    void setPrivateMining(boolean p);
```

Comment:
Why the miner server is aware of private mining?

Now, it is not aware if it is used from a external mining pool, or from MinerClientImpl. Maybe, we could implement a new PrivateMinerClientImpl, that use the new RSK signature.

A future implementation could  be PrivateHSMMinerClientImpl. The implementation to use depends on config (now, the use of MinerClientImpl depends on config)

Reply:
Currently, to make things as simple as possible, the MinerServer is aware and in fact it automatically generates fallback mined blocks when mining is active and difficulty is low.
I agree that in the future we can move this to the client side (in fact there was a comment of mine about using an HSM, but now the comment was removed by Martin)

Reply:
The first implementation (now replaces) was client-side